### PR TITLE
ipynb to markdown conversion

### LINKED ISSUE #98: add a table of contents object
This would probably be a ``pd.DataFrame`` with a multi-index, but I need a single object that I can query to get chapter names and links, rather than hard-coding these through-out the text. This might be a little clunky because you can't execute python code in a markdown cell, but that will be better than having broken links and outdated chapter names scattered throughout the text.

### LINKED ISSUE #125: should text be written in markdown and injected into notebooks?
I'm wondering if this would be a better way to handling diffing (see problems noted in [CONTRIBUTING.md](https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/master/CONTRIBUTING.md), hiding of code (#115), and maybe even hiding of text (for presentations).

### PR DESCRIPTION
**Pull request by gregcaporaso**
**NOT READY FOR MERGE**

This pull request converts all ipynb documents to markdown documents. The code that I used for the conversion (which is imperfect) is in a notebook [here](https://gist.github.com/gregcaporaso/7a141d6c30e621ae6351) and rendered [here](http://nbviewer.ipython.org/gist/gregcaporaso/7a141d6c30e621ae6351). 

- [x] Need to confirm that all content is present in output - had to fix this for the pairwise alignment chapter as the indentation was messed up (I used a ``#`` header in the middle of the notebook).
- [x] html build times out waiting for a slow-running cell. Need to override the [default timeout](https://github.com/jupyter/nbconvert/blob/master/nbconvert/preprocessors/execute.py#L26).
- [x] fix broken image links (see phylogenetic reconstruction chapter, and check others)
- [x] where to add "assignment" chapters... (these are now under ``exercises``)
- [x] get travis working
- [x] update ~~README~~ Reading IAB to describe build process
- [x] update Contributing.md, and link to it from Reading IAB (and maybe README)
- [x] review *Installation* section of *Reading IAB* with @ebolyen to confirm the links. Same for README.md. 
- [ ] after merge, confirm that *Edit on GitHub* links are working, and that images are embedded (all links refer to master - they should be right based on spot checks, but hard to test now) related to https://github.com/gregcaporaso/build-iab/issues/6

These commands now work for doing the build:

```bash
biab html -i /Users/caporaso/code/An-Introduction-To-Applied-Bioinformatics/book -o ./html
biab notebook -i /Users/caporaso/code/An-Introduction-To-Applied-Bioinformatics/book -o ./ipynb
```

Fixes #125.
Fixes #98.

### CODE REVIEW

**Comment by gregcaporaso** [comment]
Updated conversion code, notebooks seem to be working directly with ipymd now. 

New conversion code is:

```python
import os
ind = '/Users/caporaso/temp/An-Introduction-To-Applied-Bioinformatics-master/'
d = 'book'
!rm -r $d
os.makedirs('./%s' % d)

cmd = "ipython nbconvert %s --to markdown --output %s"

for root, dirs, files in os.walk(ind):
    out_root = root.replace(ind, "")
    for f in files:
        if '.ipynb_checkpoints' in root: continue
        fn, ext = os.path.splitext(f)
        if ext =='.ipynb':
            try:
                os.makedirs(os.path.join(d, out_root))
            except OSError:
                pass
            inf = os.path.join(root, f)
            outf = os.path.join(d, out_root, '%s.md' % fn)
            x = cmd % (inf, outf)
            y = !$x
            lines = open(outf)
            out_lines = []
            in_code = False
            for line in lines:
                line = line.strip('\n')
                if line.startswith('    '):
                    if not in_code:
                        in_code = True
                        out_lines.append('```python')
                    line = line[4:]
                    if len(line) > 0:
                        out_lines.append('>>> %s' % line)
                    else:
                        out_lines.append('...')
                else:
                    if in_code:
                        in_code = False
                        out_lines.append('```')
                    out_lines.append(line)
            if in_code:
                out_lines.append('```')
            outs = '\n'.join(out_lines)
            outs = outs.replace('\n\n\n', '\n\n')
            open(outf, 'w').write(outs)
            print "***WROTE: %s" % outf
```

**Comment by gregcaporaso** [comment]
These commands now work for doing the build:

```bash
biab html -i /Users/caporaso/code/An-Introduction-To-Applied-Bioinformatics/book -o ./html
biab notebook -i /Users/caporaso/code/An-Introduction-To-Applied-Bioinformatics/book -o ./ipynb
```

- [ ] Need to confirm that all content is present in output - had to fix this for the pairwise alignment chapter as the indentation was messed up (I used a ``#`` header in the middle of the notebook).

**Comment by gregcaporaso** [comment]
Updated conversion notebooks is [here](https://gist.github.com/gregcaporaso/7a141d6c30e621ae6351) and rendered [here](http://nbviewer.ipython.org/gist/gregcaporaso/7a141d6c30e621ae6351).

File: README.md [hunk 1]
```diff
@@ -1,87 +1,28 @@
...
+<div style="float: right; margin-left: 30px;"><img title="Logo by @gregcaporaso." style="float: right;margin-left: 30px;" src="https://raw.github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/to-rst/book/images/logo.png" align=right height=300/></div>
 
-* The *easiest* way is to view the static notebooks online using [nbviewer](http://nbviewer.ipython.org/). You should:
- * [start here to view the latest release version](http://nbviewer.ipython.org/github/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/0.1.0/Index.ipynb), or
- * [start here to view the development version](http://nbviewer.ipython.org/github/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/master/Index.ipynb) (which will have the latest content, but will be less polished and possibly buggy).
+IAB is in early development stage. To see the plans for where the project will go, see the [project milestones](https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/milestones). **A good example of where we're going** is the [multiple sequence alignment](http://readIAB.org/book/latest/2/3) chapter.
```
Comment (gregcaporaso) [inline README.md]: @ebolyen, does the second link on this line look possible to get to the static HTML version?

File: README.md [hunk 2]
```diff
@@ -1,87 +1,28 @@
...
-2. Applications
-  1. Studying biological diversity
+# An Introduction To Applied Bioinformatics
 
-How to use *An Introduction To Applied Bioinformatics*
-------------------------------------------------------
+An **I**ntroduction to **A**pplied **B**ioinformatics (or IAB) is a free, open source interactive text that introduces readers to core concepts of bioinformatics in the context of their implementation an application.
 
-There are two ways to use *An Introduction To Applied Bioinformatics*:
+To learn more about who should read IAB, you should begin by reading *[Reading IAB](http://readIAB.org/book/latest/1/1)*. For an overview of the content, start with the *[IAB Table of Contents](http://readIAB.org/book/latest/)*. If you're interested in getting involved in writing content or code for IAB, or editing IAB, see [CONTRIBUTING.md](https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/blob/master/CONTRIBUTING.md).
```
Comment (gregcaporaso) [inline README.md]: @ebolyen, do the first two links on this line look possible to get to the static html version?

File: book/getting-started/reading-iab.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+In the *Getting started with [Biology](alias://cf88ac) and [Computer Science](alias://6ad7e1)* sections below I provide some suggestions for other texts that will help you to get started.
+
+## How to read IAB <link src='de1bc2'/>
+
+There are two ways to read *An Introduction To Applied Bioinformatics*:
+
+* The *recommended* way is to install it and work with it interactively. See the [install instructions below](alias://96d24f).
+* The *easiest* way is to view the static notebooks online at  [readIAB.org](http://readIAB.org/). You should:
+ * [start here to view the latest release version](http://readIAB.org/book/latest/), or
```
Comment (gregcaporaso) [inline book/getting-started/reading-iab.md]: @ebolyen, does this link look possible?

File: book/getting-started/reading-iab.md [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+ * [start here to view the development version](http://readIAB.org/book/dev/) (which will have the latest content, but will be less polished and possibly buggy).
```
Comment (gregcaporaso) [inline book/getting-started/reading-iab.md]: @ebolyen, does this link look possible?

File: book/getting-started/reading-iab.md [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+IAB is split into four different sections: *Getting started*, *Fundamentals*, *Applications*, and *Wrapping up*. You should start reading IAB by working through the *Getting started* and *Fundamentals* chapters in order. You should then read the *Applications* chapters and *Wrapping up* in any order, based on your own interest.
+
+## Installation <link src='96d24f'/>
+
+If you're going to read *An Introduction to Applied Bioinformatics* interactively (recommended), you'll need to install it. The following commands should work for Linux and Mac OS X users:
+
+```
+pip install numpy
+pip install https://github.com/gregcaporaso/An-Introduction-To-Applied-Bioinformatics/archive/0.1.0.tar.gz
+wget http://readIAB.org/downloads/0.1.0/iab-notebooks-0.1.0.tgz
```
Comment (gregcaporaso) [inline book/getting-started/reading-iab.md]: @ebolyen, does this link look possible?

File: README.md (hunk 2 above)
Comment (ebolyen) [inline README.md]: No, these need to be replaced with the SHA `alias://SHA`

**Comment by gregcaporaso** [comment]
@ebolyen, I just pinged you in a few places where I'm adding links to static html or downloadable content. We can discuss in the morning, just wanted to note them now so I don't forget which ones we need to review.

File: README.md (hunk 2 above)
Comment (ebolyen) [inline README.md]: Sorry misunderstood what this file was

Comment (ebolyen) [inline README.md]: Yes, we can set the MIME-type on the S3 response to be `text/html` so the extension isn't needed!

File: README.md (hunk 1 above)
Comment (ebolyen) [inline README.md]: yep!

**Pull request merged**